dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -4,12 +4,21 @@ class Api::V1::Pd::ApplicationsController < ::ApplicationController
   # This must be included after load_and_authorize_resource so the auth callback runs first
   include Api::CsvDownload
 
-  # GET /api/v1/pd/applications
+  # GET /api/v1/pd/applications?regional_partner=:regional_partner
   def index
+    regional_partner_value = params[:regional_partner]
     application_data = empty_application_data
 
     ROLES.each do |role|
-      get_applications_by_role(role).group(:status).count.each do |status, count|
+      apps = get_applications_by_role(role).group(:status)
+      if regional_partner_value && regional_partner_value != "all" && regional_partner_value != "null"
```

Comment:
Why the string "null"?

I'm a bit confused about what this value represents. At the very least some comments would be helpful, and the "special" values (all, null, unmatched) should be constants. If we want a special value to represent no selection, I'd prefer something like "none" over "null" so it's less likely to be mixed up with actual `null` (which presumably here means no selection and is distinct from `"null"`?)

Reply:
The string "null" is what is passed when the regional partner dropdown is cleared using the little "X" button. But I should handle that in the JS and not have this controller worry about that - I'll clean this up.